### ISSUE TITLE
[BUG] IllegelMonthError even though all month values are between 1 and12

### ISSUE THREAD

**Issue by rajatscibi** [pycaret/pycaret#2396]
**Describe the bug**

I am getting "IllegalMonthError: bad month number nan; must be 1-12" even though all the dates are between 1 and 12 and no rows are missing.

**To Reproduce**

```python
import pandas as pd
df = pd.read_csv('dataset_ar.csv')

data = df.sample(frac=0.8, random_state=78603)
data_unseen = df.drop(data.index)

data.reset_index(drop=True, inplace=True)
data_unseen.reset_index(drop=True, inplace=True)

print('Data for Modeling: ' + str(data.shape))
print('Unseen Data For Predictions: ' + str(data_unseen.shape))

from pycaret.classification import *
exp_mclf101 = setup(data = data, target = 'CDO-Days', session_id=123) 
```

```python
IllegalMonthError: bad month number nan; must be 1-12
[dataset_ar.csv](https://github.com/pycaret/pycaret/files/8461328/dataset_ar.csv)
```

**Expected behavior**

**Additional context**

data file is attached for reproducing the error
[dataset_ar.csv](https://github.com/pycaret/pycaret/files/8461334/dataset_ar.csv)

**Versions**
<details>

</details>

**Comment by ngupta23** [comment]
So I see you are trying to detect the number of days since billing, but what features are you going to use? You only have 2 columns out of which 1 column only has a single value so can not be used as a feature. The other one is just a date. So what exactly is the end goal?

**Comment by rajatscibi** [comment]
Actually I've just shared few columns with you to see the month error.
There are 27 more features that are confidential.

**Comment by rajatscibi** [comment]
The column with single value is also delibrate since actual data in that
column is also confidential.

**Comment by ngupta23** [comment]
I am able to reproduce this and seems to be coming from the preprocessing code. For now, you can do this to make it work

```python
...
print('Data for Modeling: ' + str(data.shape))
print('Unseen Data For Predictions: ' + str(data_unseen.shape))

# Add this line to your code
data["Invoice Date"] = pd.to_datetime(data["Invoice Date"])

from pycaret.classification import *
exp_mclf101 = setup(data = data, target = 'CDO-Days', session_id=123) 
```

**Comment by ngupta23** [comment]
By the way, this issue will be resolved in `pycaret 3.0.0`. To test it, I used `pycaret-ts-alpha` which is currently equivalent to `pycaret 3.0.0rc`. In that version, it works without having to make that change.

https://nbviewer.org/gist/ngupta23/ca09acb51f7bd6cb29d772b7fb86b501

**Comment by rajatscibi** [comment]
nice thanks
